core/templates/dev/head/components/forms/schema_editors/SchemaBasedHtmlEditorDirective.js

```diff
@@ -28,6 +30,200 @@ oppia.directive('schemaBasedHtmlEditor', [
+                // Convert base64/URLEncoded data component to raw binary data
+                // held in a string.
+                var byteString = atob(dataURI.split(',')[1]);
+
+                // Separate out the mime component.
+                var mime = dataURI.split(',')[0].split(':')[1].split(';')[0];
+
+                // Write the bytes of the string to a typed array.
+                var ia = new Uint8Array(byteString.length);
+                for (var i = 0; i < byteString.length; i++) {
+                  ia[i] = byteString.charCodeAt(i);
+                }
+
+                var blob = new Blob([ia], {type: mime});
+                if (blob.type.match('image') &&
+                    blob.size > 0) {
+                  return blob;
+                } else {
+                  return null;
+                }
+              };
+
+              var getTrustedResourceUrlForImageFileName = function(
+                  imageFileName) {
+                var encodedFilepath = window.encodeURIComponent(imageFileName);
+                return $sce.trustAsResourceUrl(
+                  '/imagehandler/' + $scope.explorationId +
+                  '/' + encodedFilepath);
+              };
+
+              $scope.calculateTargetImageDimensions = function() {
+                var width = $scope.data.metadata.actualWidth;
+                var height = $scope.data.metadata.actualHeight;
+                if (width > OUTPUT_IMAGE_MAX_WIDTH_PX) {
+                  var aspectRatio = width / height;
+                  width = OUTPUT_IMAGE_MAX_WIDTH_PX;
+                  height = width / aspectRatio;
+                }
+                return {
+                  width: Math.round(width * $scope.imageResizeRatio),
+                  height: Math.round(height * $scope.imageResizeRatio)
+                };
+              };
+
+              $scope.setSavedImageFilename = function(filename, updateParent) {
+                $scope.data = {
+                  mode: MODE_SAVED,
+                  metadata: {
+                    savedImageFilename: filename,
+                    savedImageUrl: getTrustedResourceUrlForImageFileName(
+                      filename)
+                  }
+                };
+                if (updateParent) {
+                  AlertsService.clearWarnings();
+                  $scope.$parent.value = filename;
+                }
+              };
+
+              $scope.saveDroppedImage = function() {
+                newImageFile = convertImageDataToImageFile(addedImgNode.src);
+
+                $scope.data.metadata.uploadedFile = newImageFile;
+                $scope.data.metadata.uploadedImageData = addedImgNode.src;
+                $scope.data.metadata.actualWidth = addedImgNode.naturalWidth;
+                $scope.data.metadata.actualHeight = addedImgNode.naturalHeight;
+
+                var dimensions = $scope.calculateTargetImageDimensions();
+                var resampledImageData = getResampledImageData(
```

Comment:
What is this variable used for? I don't see it used again in a quick Ctrl-F of the code.